javaparser-core/src/main/java/com/github/javaparser/ast/expr/Name.java

```diff
@@ -238,4 +238,23 @@ public boolean replace(Node node, Node replacementNode) {
         }
         return super.replace(node, replacementNode);
     }
+
+    /**
+     * A top level name is a name that is not contained in a larger Name instance.
+     */
+    public boolean isTopLevel() {
+        return !isInternal();
+    }
+
+    /**
+     * An internal name is a name that constitutes a part of a larger Name instance.
+     */
+    public boolean isInternal() {
+        return getParentNode()
+                .filter(parent -> parent instanceof Name)
```

Comment:
I used my powers to push a rewritten version of this method :-)

Reply:
Good idea!